src/pages/DisplayFixtures.jsx

```diff
@@ -122,14 +112,16 @@ const DisplayFixtures = () => {
           },
           body: JSON.stringify(update),
         });
-        setAllPredictions([...predictions, prediction]);
-        setEditMode(false);
```

Comment:
remember, you're still in the loop here. If you `setEditMode(false)` here, that means the edit mode will be deactivated after the first loop (while we may still have more loops to go/other predictions to update), which is not what we want.

We want all changes to be processed first, then we `setEditMode` to `false` to avoid UX confusions